### PR TITLE
XEP-0045: Add <x/> tag to MUC-PMs

### PR DESCRIPTION
**Pull request by ge0rg**
To faciliate better interop with Carbons, MUC-PMs need to contain an &lt;x/&gt; tag ([rationale on standards](https://mail.jabber.org/pipermail/standards/2017-January/032048.html)).

This one is a complement to #434.

### CODE REVIEW

**Comment by intosi** [comment]
The examples have unclosed x elements.

**Comment by ge0rg** [comment]
@intosi good catch, thanks. Fixed the PR.

**Comment by SamWhited** [comment]
Added to the council agenda. Thanks. https://trello.com/c/wF37u9DJ

**Comment by iNPUTmice** [comment]
From the council chat room:

```
[06:59:09 PM] ‎Kev‎: It renders existing implementations non-compliant, and poses a potential interop problem.
‎[06:59:43 PM] ‎SamWhited‎: It is a SHOULD, not a MUST (but I'm leaning that way too)
‎[07:00:00 PM] ‎Kev‎: SamWhited: A SHOULD means you can rely on it, pretty much, because anyone not doing it made a deliberate decision.
‎[07:00:16 PM] ‎SamWhited‎: Kev: What I heard there is "you can't rely on it" :)
‎[07:00:28 PM] ‎Kev‎: SamWhited: Then it should be MAY.
‎[07:00:31 PM] ‎Kev‎: SHOULD means you can rely on it.
‎[07:00:50 PM] ‎Kev‎: At the least it needs text saying that while a service/client SHOULD attach it, receiving entities can't assume that it'll be present, because $REASONS.
‎[07:03:04 PM] ‎Kev‎: I think that an additional sentence saying something approximately "Note that this requirement was added late in MUC's life, when implementations were already stable, and thus clients MUST NOT rely on receiving this element".
‎[07:03:19 PM] ‎Kev‎: would probably be sufficient to get this text in there, while not breaking anything.
‎[07:03:20 PM] ‎SamWhited‎: yah, that seems sane
‎[07:03:27 PM] ‎daniel‎: i think everyone (including the author of the PR) will probably be fine with MAY
‎[07:03:33 PM] ‎SamWhited‎: or that
‎[07:03:54 PM] ‎Kev‎: MAY would not introduce an interop concern, so I'm far less opposed to it.
```

(shortened to be more to the point)

TLDR either change the wording to MAY or add a sentence that implementations MUST NOT rely on it.

**Comment by ge0rg** [comment]
Thanks very much for the heads up. I'll prepare an update, following the second route (because otherwise, we'll never get this sorted out).

**Comment by ge0rg** [comment]
I've added the following note to the XEP:

> Note: because this requirement was only added in revision 1.28 of this XEP, receiving clients MUST NOT rely on the existence of the <x/> element on private messages for proper processing.

**Comment by SamWhited** [comment]
@ge0rg can I get you to add a revision block to this and then we'll merge? Thanks

**Comment by ge0rg** [comment]
@SamWhited at your service :)

File: xep-0045.xml [hunk 1]
```diff
@@ -46,6 +46,14 @@
   <registry/>
   &stpeter;
   <revision>
+    <version>1.28</version>
+    <date>2017-05-31</date>
+    <initials>gl</initials>
+    <remark>
+      <p>Introduce &lt;x/> tag in MUC-PMs to support better Carbon delivery.</p>
```
Comment (SamWhited) [inline xep-0045.xml]: Nit: `&gt;` for consistency with the other side of the tag and other places tags are used in the XML

**Comment by ge0rg** [comment]
@samwhited feel free to change, I'm away from computers for the next two weeks.

**Comment by ge0rg** [comment]
Force-pushed a minor update of the revision block, to fix comment https://github.com/xsf/xeps/pull/436#pullrequestreview-41815570

**Comment by horazont** [comment]
Removing "Needs Council", apparently it got accepted [(source)](https://trello.com/c/wF37u9DJ/).
